File: tests/fixtures/typescript_simple/repository.ts
import { User, UserFilter } from './models';

export class UserRepository {
    private users: Map<number, User> = new Map();

    findById(id: number): User | undefined {
        return this.users.get(id);
    }

    findAll(): User[] {
        return Array.from(this.users.values());
    }

    save(user: User): void {
        this.users.set(user.id, user);
    }

    delete(id: number): boolean {
        return this.users.delete(id);
    }

    findByFilter(filter: UserFilter): User[] {
        let results = this.findAll();

        if (filter.role) {
            results = results.filter(u => u.role === filter.role);
        }

        if (filter.active !== undefined) {
            results = results.filter(u => u.active === filter.active);
        }

        if (filter.search) {
            const search = filter.search.toLowerCase();
            results = results.filter(u =>
                u.name.toLowerCase().includes(search)
            );
        }

        return results;
    }

    count(): number {
        return this.users.size;
    }

    exists(id: number): boolean {
        return this.users.has(id);
    }
}

File: tests/fixtures/python_simple/repository.py
from typing import Optional
from models import Item


class ItemRepository:
    def __init__(self):
        self._store: dict[int, Item] = {}

    def find_by_id(self, item_id: int) -> Optional[Item]:
        return self._store.get(item_id)

    def find_all(self) -> list[Item]:
        return list(self._store.values())

    def save(self, item: Item) -> None:
        self._store[item.id] = item

    def delete(self, item_id: int) -> bool:
        if item_id in self._store:
            del self._store[item_id]
            return True
        return False

    def find_by_category(self, category: str) -> list[Item]:
        return [item for item in self._store.values() if item.category == category]

    def count(self) -> int:
        return len(self._store)

    def exists(self, item_id: int) -> bool:
        return item_id in self._store

    def find_by_name(self, name: str) -> Optional[Item]:
        for item in self._store.values():
            if item.name == name:
                return item
        return None

    def clear(self) -> None:
        self._store.clear()
